### PR TITLE
Implementation of audit service

### LINKED ISSUE #14: Centralized audit service implementation

### PR DESCRIPTION
**Pull request by harshadk-sourcefuse**
## Description
Audit Service containing the API's for 
* save audit log
* get all audit logs
* get audit log by id
* get total count of audit log

And Test cases for Controller of above API's

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [X] Performed a self-review of my own code
- [X] npm test passes on your machine
- [X] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	@Modifying
+	void softDelete(T entity);
+
+	@Modifying
+	void softDelete(Iterable<? extends T> entities);
+
+	@Modifying
+	void softDeleteAll();
+
+//	@Modifying
```
Comment (shadab-sf) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: Remove un-necessary code

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	@Override
+	@Transactional
+	public void softDeleteAll() {
+		for (T entity : findAllActive())
+			softDelete(entity);
+	}
+
+//	@Override
```
Comment (shadab-sf) [inline libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepositoryImpl.java]: remove commented code

File: libs/core/src/test/java/com/sourcefuse/jarc/core/AuditLogEntityListenerNegativeTests.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	/**
+	 * should not save AuditLog when Role is deleted since authentication is not
+	 * exists
+	 */
+	@Test
+	void shouldNotSaveAuditLogWhenRoleIsDeleted() {
+		Role role = new Role();
+		role.setName("ABC");
```
Comment (shadab-sf) [inline libs/core/src/test/java/com/sourcefuse/jarc/core/AuditLogEntityListenerNegativeTests.java:104]: Have a meaningful rolename

File: libs/core/src/test/java/com/sourcefuse/jarc/core/AuditLogEntityListenerNegativeTests.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+		List<AuditLog> auditLogs = entityManager
+				.createQuery("Select a from AuditLog a order by a.actedOn desc", AuditLog.class).getResultList();
+
+		assertEquals(auditLogs.size(), 0);
+	}
+
+	@Test
+	void shouldSaveAuditLogWhenRoleIsSoftDeleted() {
+		Role role = new Role();
```
Comment (shadab-sf) [inline libs/core/src/test/java/com/sourcefuse/jarc/core/AuditLogEntityListenerNegativeTests.java:120]: Good candidate for code re-usability here with shouldNotSaveAuditLogWhenRoleIsDeleted()

File: services/auth-service/src/main/java/com/sourcefuse/jarc/services/authservice/AuthServiceApplication.java [hunk 5]
```diff
@@ -6,8 +6,7 @@
 @SpringBootApplication
 public class AuthServiceApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(AuthServiceApplication.class, args);
-	}
-
+  public static void main(String[] args) {
+    SpringApplication.run(AuthServiceApplication.class, args);
```
Comment (samarpan-b) [inline services/auth-service/src/main/java/com/sourcefuse/jarc/services/authservice/AuthServiceApplication.java]: why this is here ?

File: services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+  void clearUserAndAuditLog() {
+    this.clearTables();
+    this.addDefaultLogs();
+  }
+
+  @Test
+  void testCreateAuditLog() throws Exception {
+    this.mockMvc.perform(
+        post("/audit_logs")
+          .contentType("application/json")
```
Comment (mlakdawala-sf) [inline services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java]: Can we create an enum for content type in core to be consumed across the packages

File: services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+  @BeforeEach
+  void clearUserAndAuditLog() {
+    this.clearTables();
+    this.addDefaultLogs();
+  }
+
+  @Test
+  void testCreateAuditLog() throws Exception {
+    this.mockMvc.perform(
+        post("/audit_logs")
```
Comment (mlakdawala-sf) [inline services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java]: Can we create a constant for base path as well in the file

**Review by mlakdawala-sf — CHANGES_REQUESTED** [review]
Rest looks good

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java (hunk 1 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepository.java]: done

File: libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepositoryImpl.java (hunk 2 above)
Comment (harshadk-sourcefuse) [reply libs/core/src/main/java/com/sourcefuse/jarc/core/softdelete/SoftDeletesRepositoryImpl.java]: done

File: services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java (hunk 6 above)
Comment (harshadk-sourcefuse) [reply services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java]: Done

File: services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java (hunk 7 above)
Comment (harshadk-sourcefuse) [reply services/audit-service/src/test/java/com/sourcefuse/jarc/services/auditservice/AuditControllerTests.java]: Done

**Review by samarpan-b — APPROVED** [review]

**Review by mlakdawala-sf — APPROVED** [review]

**Pull request merged**